### PR TITLE
Pxapi2/feature/pbi 155250 codelist outputvalues

### PR DESCRIPTION
**Pull request by MikaelNordberg**
Load aggregations for PX databases.
Implemented codelist and outputValues when getting selection.

### CODE REVIEW

File: PxWeb/Code/Api2/DataSelection/ISelectionHandler.cs [hunk 1]
```diff
@@ -5,7 +5,6 @@ namespace PxWeb.Code.Api2.DataSelection
 {
     public interface ISelectionHandler
     {
-        public bool Verify(PXModel model, VariablesSelection? variablesSelection, out Problem? problem);
-        public Selection[] GetSelection(PXModel model, VariablesSelection? variablesSelection);
+        public Selection[]? GetSelection(IPXModelBuilder builder, PXModel model, VariablesSelection? variablesSelection, out Problem? problem);
```
Comment (likp) [inline PxWeb/Code/Api2/DataSelection/ISelectionHandler.cs:8]: You should probably remove the model from the arguments and use to model on the builder

File: PxWeb/Code/Api2/DataSelection/SelectionHandler.cs [hunk 2]
```diff
@@ -28,50 +33,80 @@ public class SelectionHandler : ISelectionHandler
...
             problem = null;
-
+            
             if (variablesSelection is not null && HasSelection(variablesSelection))
             {
                 //Verify that variable exists
                 foreach (var variable in variablesSelection.Selection)
                 {
-                    if (!model.Meta.Variables.Any(x => x.Code.ToUpper().Equals(variable.VariableCode.ToUpper())))
+                    Variable? pxVariable = model.Meta.Variables.FirstOrDefault(x => x.Code.ToUpper().Equals(variable.VariableCode.ToUpper()));
```
Comment (likp) [inline PxWeb/Code/Api2/DataSelection/SelectionHandler.cs:98]: Instead of ToUpper do a case insensitive comapre

**Review by likp — COMMENTED** [review]
Some minro adjustments to do

File: PxWeb/Code/Api2/DataSelection/ISelectionHandler.cs (hunk 1 above)
Comment (MikaelNordberg) [reply PxWeb/Code/Api2/DataSelection/ISelectionHandler.cs]: Now uses builder.Model for the PxModel

File: PxWeb/Code/Api2/DataSelection/SelectionHandler.cs (hunk 2 above)
Comment (likp) [reply PxWeb/Code/Api2/DataSelection/SelectionHandler.cs]: I was thiniking more in line with `string.Compare(x.Code, variable.VariableCode, StringComparison.InvariantCultureIgnoreCase) == 0` otherwiase we are creating allot of strings just for the comparison.

**Review by likp — APPROVED** [review]

**Pull request merged**